plugins/crypto/src/main/java/org/opensearch/crypto/CryptoStore.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.crypto;
+
+import com.amazonaws.encryptionsdk.CryptoAlgorithm;
+import com.amazonaws.encryptionsdk.caching.CachingCryptoMaterialsManager;
+import com.amazonaws.encryptionsdk.caching.LocalCryptoMaterialsCache;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.io.InputStreamContainer;
+import org.opensearch.common.unit.ByteSizeUnit;
+import org.opensearch.common.unit.ByteSizeValue;
+import org.opensearch.common.util.concurrent.AbstractRefCounted;
+import org.opensearch.crypto.cache.MasterKeyCache;
+import org.opensearch.encryption.CryptoUtils;
+import org.opensearch.encryption.core.AwsCrypto;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+
+import java.io.InputStream;
+import java.security.SecureRandom;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class CryptoStore extends AbstractRefCounted implements CryptoClient {
```

Comment:
can this also be moved to core ? Key provider base plugin shouldn't be containing this.

Reply:
Made structural changes as discussed here https://github.com/opensearch-project/OpenSearch/pull/8465#discussion_r1297449003